formats/hocon/src/test/kotlin/kotlinx/serialization/hocon/HoconRootObjectsTest.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.hocon
+
+import com.typesafe.config.ConfigFactory
+import kotlinx.serialization.Serializable
+import org.junit.Assert.*
+import org.junit.Ignore
+import org.junit.Test
+
+class HoconRootMapTest {
+    private val configForRootMap = """
+        key1 {
+            a = "text1"
+            b = 11
+            }
+        
+        key2 {
+            a = "text2"
+            b = 12
+        }
+        
+        key3 {
+            a = "text3"
+            b = 13
+        }
+    """
+    private val configWithEmptyObject = "{}"
+    private val configWithRootList = "[foo, bar]"
+    private val emptyConfig = ""
+
+    @Serializable
+    data class CompositeValue(
+            val a: String,
+            val b: Int
+    )
+
+    @Test
+    fun testConfigWithRootMap() {
+        val config = ConfigFactory.parseString(configForRootMap)
+        val obj = Hocon.decodeFromConfig<Map<String, CompositeValue>>(config)
+
+        assertEquals(CompositeValue("text1", 11), obj["key1"])
+        assertEquals(CompositeValue("text2", 12), obj["key2"])
+        assertEquals(CompositeValue("text3", 13), obj["key3"])
+    }
+
+    @Test
+    fun testEmptyObjectDecode() {
+        val config = ConfigFactory.parseString(configWithEmptyObject)
+        //non-null map decoded from empty object as empty map
+        val map = Hocon.decodeFromConfig<Map<String, String>>(config)
+        assertTrue(map.isEmpty())
+
+        //nullable map decoded from empty object as null - not obvious
+        val nullableMap = Hocon.decodeFromConfig<Map<String, String>?>(config)
+        assertNull(nullableMap)
+
+        //root-level list in config not supported but nullable list can be decoded from empty object
+        val nullableList = Hocon.decodeFromConfig<List<String>?>(config)
+        assertNull(nullableList)
```

Comment:
It seems there's no need for separate variable and you can join lines: `assertNull(Hocon.decodeFromConfig<List<String>?>(config))`